File: testutils-logrecorder-logback/src/main/java/info/novatec/testit/testutils/logrecorder/junit5/RecordLoggers.java
package info.novatec.testit.testutils.logrecorder.junit5;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.junit.jupiter.api.extension.ExtendWith;


/**
 * Annotating any test method with this annotation will register the {@link LogbackRecorderExtension} with JUnit 5.
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@ExtendWith(LogbackRecorderExtension.class)
public @interface RecordLoggers {

    /**
     * @return classes who's name should be used to identify the loggers to record.
     */
    Class<?>[] value() default {};

    /**
     * @return names of the loggers to record.
     */
    String[] names() default {};

}

File: logrecorder/logrecorder-logback/src/main/kotlin/info/novatec/testit/logrecorder/logback/junit5/RecordLoggers.kt
package info.novatec.testit.logrecorder.logback.junit5

import org.junit.jupiter.api.extension.ExtendWith
import kotlin.reflect.KClass

/**
 * Annotating any test method with this annotation will start the recording of
 * log messages while the test method is executed.
 *
 * Only those loggers configured with this annotation - either by it's class or
 * name - will be recorded! Multiple loggers can be set by either method and
 * combinations between class and explicit names are possible as well.
 *
 * @see value
 * @see names
 * @since 1.0
 */
@Retention
@Target(AnnotationTarget.FUNCTION)
@ExtendWith(LogbackRecorderExtension::class)
annotation class RecordLoggers(

    /**
     * Classes who's name should be used to identify loggers to record.
     */
    vararg val value: KClass<*> = [],

    /**
     * Names of loggers to record.
     */
    val names: Array<String> = []

)
